app/uk/gov/hmrc/agentmappingfrontend/controllers/MappingController.scala

```diff
@@ -69,108 +68,52 @@ class MappingController @Inject()(
     }
   }
 
-  def startSubmit: Action[AnyContent] = Action.async { implicit request =>
-    successful(Redirect(routes.MappingController.showEnterAccountNo).withNewSession)
-  }
-
-  def showEnterAccountNo: Action[AnyContent] = Action.async { implicit request =>
-    withAuthorisedAgentAudited { providerId =>
-      successful(Ok(html.enter_account_number(mappingFormArn)))
-    }(AuditService.auditCheckAgentRefCodeEvent(auditService))
-  }
-
-  def submitEnterAccountNo: Action[AnyContent] = Action.async { implicit request =>
-    withAuthorisedAgent { providerId =>
-      mappingFormArn.bindFromRequest.fold(
-        formWithErrors => {
-          successful(Ok(html.enter_account_number(formWithErrors)))
-        },
-        formArn => {
-          successful(Redirect(routes.MappingController.showEnterUtr).addingToSession(("mappingArn", formArn.arn.value)))
+  def startSubmit(id: MappingArnResultId): Action[AnyContent] = Action.async { implicit request =>
+    withAuthorisedAgent(id) { _ =>
+      repository.findArn(id).flatMap {
+        case Some(arn) => {
+          for {
+            newRefForArn <- repository.create(arn)
+            doMappingResult <- mappingConnector.createMapping(arn).map {
+                                case CREATED =>
+                                  Redirect(routes.MappingController.complete(newRefForArn))
+                                    .withSession(request.session + ("mappingArn" -> arn.value))
```

Comment:
There may be some potential for a future refactor here to make things a tad simpler.
The ARN is stored in the session here, but we're also storing it and creating a new ID here too and then passing that ID through to all of the views and then from those views straight back to our `SignedOutController.signOutAndRedirect` controller endpoint again.
It looks like we could just store the ARN in the session (or keystore) until our `SignedOutController.signOutAndRedirect` endpoint is called, whereupon we'd have the code to pull out the ARN from the session/keystore, store it in the repository and create an ID for it, then redirect to GG sign in with the ID.
One for a small tech debt maybe?

Reply:
the arn stored here is simply for verificaiton only, so nobody can land on the complete page unless they have a key = "mappingArn" with a valid Arn before the complete page is visible for them, to prevent accidental view as the mapping journey is very sudden, 1 page and you're done. This should assure they actually have completed the mapping journey before they can see the complete page for the most recent logged in gg account.